### PR TITLE
Enhance/#8177 - Add the settings section toggle switch for audience segmentation widget area visibility

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #8177 

## Relevant technical choices

* The IB suggests creating a new file in the `assets/js/modules/analytics-4/components/settings/AudienceSegmentation` directory. However, as a convention, all other admin settings files and components are in `assets/js/components/settings/`. Hence, the `SettingsVisitorGroups` component is created in `assets/js/components/settings/`.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by tofumatt — APPROVED** [review]

File: assets/js/components/settings/SettingsCardVisitorGroups.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			fill
+			rounded
+		>
+			<div className="googlesitekit-settings-module googlesitekit-settings-module--active">
+				<Grid>
+					<Row>
+						<Cell size={ 12 }>
+							<Switch
+								label={ __(
+									'Display visitors groups in dashboard',
```
Comment (aaemnnosttv) [inline assets/js/components/settings/SettingsCardVisitorGroups.js:68]: This is a misspelling which was copied from the design. I've raised it there as well but we should fix this in a follow-up.

**Review by aaemnnosttv — COMMENTED** [review]
@techanvil @hussain-t

File: assets/js/components/settings/SettingsCardVisitorGroups.js (hunk 1 above)
Comment (techanvil) [reply assets/js/components/settings/SettingsCardVisitorGroups.js:68]: Thanks @aaemnnosttv, I've created https://github.com/google/site-kit-wp/issues/8496 to address this.

File: assets/js/components/settings/SettingsCardVisitorGroups.js
Comment (techanvil) [inline assets/js/components/settings/SettingsCardVisitorGroups.js:1]: Hi @hussain-t, just a heads up that while I did spot your note about the location for this file in **relevant technical choices**, the location specced in the IB was actually correct, in line with the [code organization](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.mlj74m7pz9gr) section of the design doc and as mentioned here: https://github.com/google/site-kit-wp/issues/8177#issuecomment-1926819726

An extra wrinkle is that in fact we have also identified this initial code org as needing improvement and we're going to move the Audience Segmentation component files under `assets/js/modules/analytics-4/components/audience-segmentation/` instead.

I've created https://github.com/google/site-kit-wp/issues/8495 to address the wider reorganisation, but in the meantime we should stick to the code structure outlined in the design doc to make it easy to keep track of all the Audience Segmentation components and move them appropriately when the time comes.

I've also created https://github.com/google/site-kit-wp/issues/8496 to address the typo that Evan spotted and mentioned separately in this issue (which of course was simply copied from Figma and not introduced via this PR), and to move this component at the same time.

Comment (hussain-t) [reply assets/js/components/settings/SettingsCardVisitorGroups.js:1]: Thanks for pointing it out, @techanvil 👍

**Pull request merged**